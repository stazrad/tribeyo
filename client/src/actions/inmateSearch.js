import fetch from 'isomorphic-fetch'

export const autocomplete = input => {
	// return empty predictions array if input is blank
	if (!input) {
		return dispatch =>
			dispatch({
				type: 'AUTOCOMPLETE',
				predictions: []
			})
	}
	return dispatch => {
		const config = {
			headers: {
				'Content-Type': 'application/json'
			},
			mode: 'same-origin',
			credentials: 'include'
		}
		fetch(`/api/autocomplete/${input}`, config)
			.then(res => res.json())
			.then(predictions => {
				return dispatch({
					type: 'AUTOCOMPLETE',
					predictions
				})
			})
			.catch(err => {
				const error = {
					type: 'server',
					message: 'Oops! Something went wrong...try again.'
				}
				console.log(err)
			})
	}
}

export const searchByInmate = name => dispatch => {
	const config = {
		headers: {
			'Content-Type': 'application/json'
		},
		mode: 'same-origin',
		credentials: 'include'
	}
	fetch(`/api/searchByInmate?nameFirst=${name.first}&nameLast=${name.last}`, config)
		.then(response => response.json())
		.then(predictions => {
			return dispatch({
				type: 'INMATE_RESULTS',
				predictions
			})
		})
		.catch(err => {
			const error = {
				type: 'server',
				message: 'Oops! Something went wrong...try again.'
			}
			console.log(err)
		})
}
