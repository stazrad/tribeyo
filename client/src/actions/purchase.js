export const purchaseStart = () => {
    return (dispatch) => (
        type: 'PURCHASE_START',
        purchaseProcess
    )
}

export const purchaseEnd = () => {
    return (dispatch) => (
        type: 'PURCHASE_END',
        purchaseProcess
    )
}
