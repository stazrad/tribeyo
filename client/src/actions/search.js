// PACKAGES //
import fetch from 'isomorphic-fetch'

export const autocomplete = input => {
    // return empty predictions array if input is blank
    if(!input) {
        return(dispatch) => dispatch({
            type: 'AUTOCOMPLETE',
            predictions: []
        })
    }
    return (dispatch) => {
        const init = {
            headers: {
                'Content-Type': 'application/json'
            }
        }
        fetch(`/api/autocomplete/${input}`, init)
            .then(response => response.json())
            .then(predictions => {
                return dispatch({
                    type: 'AUTOCOMPLETE',
                    predictions
                })
            })
            .catch(err => {
                const error = {
                    type: 'server',
                    message: 'Oops! Something went wrong...try again.'
                }
                console.log(err)
            })
    }
}

export const searchByCity = input => {
    return (dispatch) => {
        const init = {
            headers: {
                'Content-Type': 'application/json'
            }
        }
        fetch(`/api/searchByCity/${input}`, init)
            .then(response => response.json())
            .then(areaCode => {
                return dispatch({
                    type: 'SEARCH_BY_CITY',
                    areaCode
                })
            })
            .catch(err => {
                const error = {
                    type: 'server',
                    message: 'Oops! Something went wrong...try again.'
                }
                console.log(err)
            })
    }
}
