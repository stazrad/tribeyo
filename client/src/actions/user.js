// packages
import fetch from 'isomorphic-fetch'

const login = user => ({
	type: 'LOGIN',
	user
})

const loginError = error => ({
	type: 'LOGIN_ERROR',
	error
})

const signupError = error => ({
	type: 'SIGNUP_ERROR',
	error
})

export const userLogin = formData => dispatch => {
	const init = {
		headers: {
			'Content-Type': 'application/json'
		},
		method: 'POST',
		body: JSON.stringify(formData)
	}
	fetch('/api/profile/login', init)
		.then(res => res.json())
		.then(res => {
			if (res.status != 200) {
				const msg = res.msg.toLowerCase()
				const error = {}

				if (msg.includes('email')) {
					error.type = 'email'
					error.msg = res.msg
				} else if (msg.includes('password')) {
					error.type = 'password'
					error.msg = 'This password is invalid'
				} else {
					error.type = 'server'
					error.msg = res.msg
				}

				return dispatch(loginError(error))
			} else {
				return dispatch(login(res.user))
			}
		})
		.catch(err => {
			const error = {
				type: 'server',
				msg: 'Oops! Something went wrong...try again.'
			}
			return dispatch(loginError(error))
		})
}

export const userSignup = formData => dispatch => {
	const init = {
		headers: {
			'Content-Type': 'application/json'
		},
		method: 'POST',
		body: JSON.stringify(formData)
	}
	fetch('/api/profile', init)
		.then(res => res.json())
		.then(res => {
			if (res.status != 200) {
				const msg = res.msg.toLowerCase()
				const error = {}

				if (msg.includes('email')) {
					error.type = 'email'
					error.msg = res.msg
				} else if (msg.includes('password')) {
					error.type = 'password'
					error.msg = res.msg
				} else {
					error.type = 'server'
					error.msg = res.msg
				}

				return dispatch(signupError(error))
			} else {
				return dispatch(login(res.user))
			}
		})
		.catch(err => {
			const error = {
				type: 'server',
				msg: 'Oops! Something went wrong...try again.'
			}
			return dispatch(signupError(error))
		})
}
