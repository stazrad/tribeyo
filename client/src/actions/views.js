export const checkoutView = checkout => ({
	type: 'CHECKOUT_VIEW',
	checkout
})

export const searchView = search => ({
	type: 'SEARCH_VIEW',
	search
})
