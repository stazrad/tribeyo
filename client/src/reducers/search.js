const defaultState = {
  areaCode: {
    code: '',
    display: ''
  },
  predictions: []
}

const searchReducer = (
  state = defaultState,
  { areaCode, predictions, type }
) => {
  switch (type) {
    case 'AUTOCOMPLETE':
      return {
        ...state,
        predictions
      }
    case 'SEARCH_BY_CITY':
      return {
        ...state,
        areaCode
      }
    default:
      return state
  }
}

export default searchReducer
