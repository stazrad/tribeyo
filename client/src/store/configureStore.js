// PACKAGES //
import { createStore, combineReducers, applyMiddleware, compose } from 'redux'
import thunk from 'redux-thunk'

// REDUCERS //
import userReducer from '../reducers/user'

const composeEnhancers = window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose()

export default () => {
    const store = createStore(
        combineReducers({
            user: userReducer
        }), composeEnhancers(applyMiddleware(thunk))
    )
    return store
}
