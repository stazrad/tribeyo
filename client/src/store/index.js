// packages
import { applyMiddleware, combineReducers, compose, createStore } from 'redux'
import thunk from 'redux-thunk'

// imports
import search from 'reducers/search'
import user from 'reducers/user'

const composeEnhancers =
	window.__REDUX_DEVTOOLS_EXTENSION_COMPOSE__ || compose() //TODO remove devtools in production env

export default () =>
	createStore(
		combineReducers({
			search,
			user
		}),
		composeEnhancers(applyMiddleware(thunk))
	)
