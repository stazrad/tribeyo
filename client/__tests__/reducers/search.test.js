import searchReducer, { defaultState } from 'reducers/search'
import { autocomplete } from 'actions/search'
import store from 'store'

describe('search reducer', () => {
    it('should return empty predictions array', () => {
        // test empty input
        store().dispatch(autocomplete())

        const { predictions } = store().getState().search

        expect(predictions).toEqual([])
    })

    it('should return autocomplete predictions', () => {
        const predictions = [
            'you have my sword',
            'and my bow',
            'and my axe'
        ]
        const action = {
            type: 'AUTOCOMPLETE',
            predictions
        }
        const result = searchReducer(defaultState, action)

        expect(result).toMatchSnapshot()
    })

    it('should return area code object', () => {
        const areaCode = {
            code: '615',
            display: '(615)'
        }
        const action = {
            type: 'SEARCH_BY_CITY',
            areaCode
        }
        const result = searchReducer(defaultState, action)

        expect(result).toMatchSnapshot()
    })
})
