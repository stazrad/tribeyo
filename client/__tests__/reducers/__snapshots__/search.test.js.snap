// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`search reducer should return area code object 1`] = `
Object {
  "areaCode": Object {
    "code": "615",
    "display": "(615)",
  },
  "predictions": Array [],
}
`;

exports[`search reducer should return autocomplete predictions 1`] = `
Object {
  "areaCode": Object {
    "code": "",
    "display": "",
  },
  "predictions": Array [
    "you have my sword",
    "and my bow",
    "and my axe",
  ],
}
`;
